src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -482,11 +490,11 @@ public boolean selectAttackSubs(final Territory unitTerritory) {
     final List<Unit> defenders = (List<Unit>) battle.getDefendingUnits();
     ProLogger.info(player.getName() + " checking sub attack in " + unitTerritory + ", attackers=" + attackers
         + ", defenders=" + defenders);
-    ProBattleUtils.setData(getGameData());
+    calc.setData(getGameData());
 
     // Calculate battle results
     final ProBattleResult result =
-        ProBattleUtils.calculateBattleResults(player, unitTerritory, attackers, defenders, new HashSet<Unit>(), true);
+        calc.calculateBattleResults(player, unitTerritory, attackers, defenders, new HashSet<Unit>(), true);
```

Comment:
So you only want to set the data once into the calc per phase since it copies the data. Otherwise if you set it for each calc call then it would copy the data every time.